classes/ObjectModel.php

```diff
@@ -892,17 +892,23 @@ public function delete()
 
         $this->clearCache();
         $result = true;
-        // Remove association to multishop table
-        if (Shop::isTableAssociated($this->def['table'])) {
-            $id_shop_list = Shop::getContextListShopID();
-            if (count($this->id_shop_list)) {
-                $id_shop_list = $this->id_shop_list;
-            }
 
-            $id_shop_list = array_map('intval', $id_shop_list);
+        // Remove association to multishop table
+        if (count($this->id_shop_list)) {
+            $shopIdsList = $this->id_shop_list;
+        } else {
+            $shopIdsList = Shop::getContextListShopID();
+        }
+        $shopIdsList = array_map('intval', $shopIdsList);
```

Comment:
I think we should extract this into protected method and it can be reused instead of one I recently introduced for combination https://github.com/PrestaShop/PrestaShop/blob/5ed616c94b2ab87ccaeae76cfda3e2013c29503f/classes/Combination.php#L541
wdyt?